lib/hyrax/transactions/steps/set_uploaded_date.rb

```diff
@@ -3,25 +3,44 @@ module Hyrax
   module Transactions
     module Steps
       ##
-      # A `dry-transaction` step that sets the uploaded date to now for an
-      # input work.
+      # A step that sets the uploaded date to now for an input resource or
+      # change_set
       #
       # @since 2.4.0
       class SetUploadedDate
-        include Dry::Transaction::Operation
+        include Dry::Monads[:result]
 
         ##
-        # @note the current implementation sets the uploaded date to
+        # @param [#time_in_utc] time_service
+        def initialize(time_service: Hyrax::TimeService)
+          @time_service = time_service
+        end
+
+        ##
+        # @note the implementation sets the uploaded date to
         #   `#date_modified` if it exists, falling back on the current datetime.
         #
-        # @param [Hyrax::WorkBehavior] work
+        # @param [#date_uploaded=] obj
         #
         # @return [Dry::Monads::Result]
-        def call(work)
-          work.date_uploaded =
-            work.date_modified || Hyrax::TimeService.time_in_utc
-          Success(work)
+        def call(obj)
+          return Failure[:no_date_uploaded_attribute, obj] unless
+            obj.respond_to?(:date_uploaded=)
+
+          obj.date_uploaded = date_uploaded(obj)
+
+          Success(obj)
         end
+
+        private
+
+          def date_uploaded(obj)
+            if obj.try(:date_modified).present?
+              obj.try(:date_modified)
+            else
+              @time_service.time_in_utc
```

Comment:
I don't think we have a `@time_service` object (as per  lib/hyrax/transactions/steps/set_modified_date.rb)

Reply:
i'm not following. `@time_service` is set in the initializer, right?

Reply:
You are correct. I must have overlooked the initializer.